src/BenchmarkDotNet.TestAdapter/Package/BenchmarkDotNet.TestAdapter.props

```diff
@@ -0,0 +1,12 @@
+﻿<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <!-- Disables generation of MSTest entrypoint -->
+    <GenerateProgramFile>false</GenerateProgramFile>
+    <EntryPointSourceDirectory>$(MSBuildThisFileDirectory)..\entrypoints\</EntryPointSourceDirectory>
+  </PropertyGroup>
+  <ItemGroup Condition="'$(GenerateBDNEntryPoint)' == 'true'">
```

Comment:
Do we need the condition still? I would expect the entry point to always be generated if I'm using the test adapter.

Reply:
Yes, this is still needed for projects that already have an entry point or want to use their own because of the customisations that it has. If we don't have this condition, then the compilation will fail because there are two entry points detected.

Reply:
>  for projects that already have an entry point

I don't think that's a concern. Using the test adapter is changing how the benchmarks are ran and so users should remove their own entry point. The same way you don't expect to write an entry point for unit test libraries.

> or want to use their own because of the customisations that it has

Ok, that's a valid point. But I still think it's not a concern, as advanced usages like that can force the entry point used, even if 2 exist. See https://github.com/dotnet/BenchmarkDotNet/blob/b3b2d913a9de86147167f3fe65d770969c9541a5/src/BenchmarkDotNet/Templates/CsProj.txt#L29

Reply:
Fair points, what if I just invert the default condition instead so it is auto-generated by default, with a way to disable it? Then it would behave the same as MSTest which provides a way to conditionally disable the entry point generation with `GenerateProgramFile`. Also I did not realise that was what `StartupObject` was for, that's neat.

Reply:
I think that's a fine idea to swap the default. I wonder if it can be disabled from the csproj like that, though? If the props file imports the entry point file before the csproj disables the property, it will essentially be ignored. How does MSTest do it?

Reply:
From my testing, it seems that the csproj runs before the props file is imported, which makes sense because the csproj is the one that loads the package reference. So after inverting this condition, it all works as described.

Reply:
Can we hijack `GenerateProgramFile`? Something like this?

```xml
<PropertyGroup>
    <EntryPointSourceDirectory>$(MSBuildThisFileDirectory)..\entrypoints\</EntryPointSourceDirectory>
</PropertyGroup>
<ItemGroup Condition="'$(GenerateProgramFile)' != 'false'">
    <Compile Include="$(EntryPointSourceDirectory)EntryPoint.cs" Condition="'$(Language)' == 'C#'" Visible="false"/>
    <CompileAfter Include="$(EntryPointSourceDirectory)EntryPoint.fs" Condition="'$(Language)' == 'F#'" Visible="false"/>
    <Compile Include="$(EntryPointSourceDirectory)EntryPoint.vb" Condition="'$(Language)' == 'VB'" Visible="false"/>
</ItemGroup>
<PropertyGroup>
    <!-- Disables generation of MSTest entrypoint -->
    <GenerateProgramFile>false</GenerateProgramFile>
</PropertyGroup>
```

Reply:
I managed to get it working, turns out that PropertyGroup has higher precedence than ItemGroup when outside of a Target, but when inside a Target they will be executed in order. I've moved this into a target now and it works perfectly.